marklogic-client-api/src/main/java/com/marklogic/client/impl/DocumentWriteOperationImpl.java

```diff
@@ -70,4 +81,37 @@ public DocumentMetadataWriteHandle getMetadata() {
   public AbstractWriteHandle getContent() {
     return content;
   }
+
+  @Override
+  public int compareTo(DocumentWriteOperation o) {
+    if(o == null)
+      throw new NullPointerException("DocumentWriteOperation cannot be null");
+
+    if(this.getUri() instanceof String && o.getUri() instanceof String)
+      return getUri().compareTo(o.getUri());
+
+    if(this.getUri() == null && o.getUri() instanceof String)
+      return -1;
+
+    if(this.getUri() instanceof String && o.getUri()==null)
+      return 1;
+
+    if(this.getUri() == null && o.getUri() == null)
+      super.hashCode();
+    return 0;
+  }
+
+  @Override
+  public int hashCode() {
+    if(this.getUri()!=null)
+      return this.getUri().hashCode();
+    return super.hashCode();
+  }
+
+  @Override
+  public boolean equals(Object o){
+    if(this.getUri()!=null)
+      return this.getUri().equals(o);
```

Comment:
getUri() returns a String, which can never equal a DocumentWriteOperation

Would it make sense to return

- false if o is not an instance of DocumentWriteOperation
- super.equals() on o if this.getUri() is null
- this.getUri().equals() on the uri of o after casting o to a DocumentWriteOperation